tests/datasets/test_datasets.py

```diff
@@ -0,0 +1,53 @@
+# -*- coding: utf-8 -*-
```

Comment:
There's a couple more tests we could do, I think:

- the `STELLARGRAPH_DATASETS_PATH` override too
- checking that the caching works (i.e. that it doesn't redownload if all the files are there)

Reply:
> * checking that the caching works (i.e. that it doesn't redownload if all the files are there)

I wasn't able to get this to work with pytest monkeypatch on urlretrieve; can you think of a better way?

Reply:
Maybe capturing the logging and then checking for the second code path? Could also have `download` return some status flag (e.g. `True` if it downloaded, `False` if it used the cache). Neither are that great, though.

Reply:
I managed to get unittest.mock to work wrapping urlretrieve